### ISSUE TITLE
Crash with SchemaValidationException(errors)

### ISSUE THREAD

**Issue by aakashs** [realm/realm-cocoa#3162]
Hi, I'm using RealmSwift and it has been working great but it's started to crash when I instantiate a new Realm object.

Attached is the crash log. When I clean the app before running, it tells me "object_store.hpp" is missing (which I don't think this is), but this doesn't seem to prevent the app from building.

I thought this might be a cocoapods issue but I'm using 0.39.0 and I tried clearing my cache and this didn't change anything.

Thank you in advance!

<img width="1218" alt="screenshot 2016-02-01 02 10 41" src="https://cloud.githubusercontent.com/assets/2073966/12710926/2af000f6-c889-11e5-9e14-03d47dc2db8b.png">

**Comment by aakashs** [comment]
Update:
So this error is because of these 2 lines
    let added = RealmOptional<Bool>()
    let timeSpent = RealmOptional<Int>()

Changing them to the following fixes things.
    var added: Bool?
    var timeSpent: Int?

But I'm not sure this is the right way to declare optional variables?

**Comment by jpsim** [comment]
It would help if you shared what the exception being thrown was instead of sharing a screenshot stopped at a breakpoint right before the invalid schema exception is thrown. If you `continue` from LLDB, the exception should be printed. That should tell you precisely what's incorrect with your schema.

The first way you declared your optional primitive properties was correct:

```swift
let added = RealmOptional<Bool>()
let timeSpent = RealmOptional<Int>()
```

The following properties can't be represented by the Objective-C runtime, so although it will compile, the properties won't be visible to Realm:

```swift
var added: Bool?
var timeSpent: Int?
```

Once you share the exception message, we'll be able to help you a bit more.

**Comment by aakashs** [comment]
`fatal error: 'try!' expression unexpectedly raised an error: Error Domain=io.realm Code=0 "Migration is required due to the following errors: 
- Property 'added' has been added to latest object model." UserInfo={NSLocalizedDescription=Migration is required due to the following errors: 
- Property 'added' has been added to latest object model.}: file /Library/Caches/com.apple.xbs/Sources/swiftlang/swiftlang-700.1.101.15/src/swift/stdlib/public/core/ErrorType.swift, `

This was my exception. This was my first time updating my data model so I had to run a migration in order to get it to work properly. Sorry, closing this issue.
